package NMiniYT.NTablet.NApi;

import "distributed_transactions/tablet/proto/transaction_id.proto";

// messages between transaction coordinator and participants

message TReqParticipantPrepareTransaction
{
    required TTransactionID TransactionID = 1;

    optional uint64 HybridLocalClock = 2 [default = 0];
}

message TRspParticipantPrepareTransaction
{
    required bool IsSuccess = 1;
    optional string Error = 2;

    optional uint64 HybridLocalClock = 3 [default = 0];
}

message TReqParticipantCommitTransaction
{
    required TTransactionID TransactionID = 1;
    required uint64 CommitTimestamp = 2;

    optional uint64 HybridLocalClock = 3 [default = 0];
}

message TRspParticipantCommitTransaction
{
    optional uint64 HybridLocalClock = 1 [default = 0];
}

message TReqParticipantAbortTransaction
{
    required TTransactionID TransactionID = 1;
}

message TRspParticipantAbortTransaction
{
}
